### PR TITLE
Inject image registry value into components

### PR DESCRIPTION
**Pull request by fgimenez**
Sets image registry for all the components on the configmap service and adds configmap resources for aws and kvm.

### CODE REVIEW

File: pkg/v5/configmap/desired.go [hunk 1]
```diff
@@ -11,29 +11,45 @@ import (
...
+
 func (s *Service) GetDesiredState(ctx context.Context, configMapValues ConfigMapValues) ([]*corev1.ConfigMap, error) {
 	desiredConfigMaps := make([]*corev1.ConfigMap, 0)
 
-	{
-		configMap, err := s.newIngressControllerConfigMap(ctx, configMapValues, s.projectName)
+	generators := []configMapGenerator{
+		s.newIngressControllerConfigMap,
+		s.newKubeStateMetricsConfigMap,
+		s.newNodeExporterConfigMap,
```
Comment (rossf7) [inline pkg/v5/configmap/desired.go]: I like the generator concept being introduced here. <3

**Comment by puja108** [comment]
Indeed we should move IC and maybe more things into chart-operator soon

**Comment by rossf7** [comment]
Totally agree but we also need to get monitoring implemented soon.

File: pkg/v5/configmap/desired.go (hunk 1 above)
Comment (fgimenez) [reply pkg/v5/configmap/desired.go]: Thanks! :) Here the intention is only not repeating code, but yes, maybe the concept can be useful in general.

**Pull request merged**